### ISSUE TITLE
Span-based Semantic Parsing for Compositional Generalization

### ISSUE THREAD

**Issue by Tahkare** [Tahkare/PhD-archive#4]
## TL;DR
They propose a span-based parser using an approximate CYK algorithm for compositional generalization.
A score is computed for each span/category combination.
Then the CYK algorithm iterates bottom-up to find for each span the K best trees that follow the grammar rules and, at train time, that produce a part of the gold program.
Each span is then optimized independently to help with compositional generalization.

### Article Link
https://arxiv.org/pdf/2009.06040.pdf

### Author
J. Herzig & J. Berant

## Key Takeaways
* Span-based parsing works well for compositional generalization
* Approximate CYK algorithm for span-based parsing without access to gold trees
